src/main/java/Duke.java

```diff
@@ -1,10 +1,178 @@
+            MessageManager.printHelp(commands[1]);
+            break;
+        case "delete":
+            int deleteIndex = getPositiveNumeric(commands[1]);
+            System.out.println(deleteIndex > 0 && deleteIndex <= tasks.size()
+                    ? "Noted. I've removed this task:\n " + MessageManager.taskToString(tasks.get(deleteIndex - 1)) + "\nNow you have " + (tasks.size() - 1) + " tasks in the list.\n"
+                    : "Formatting error");
+            tasks.remove(deleteIndex - 1);
+            writeIntoFile();
+            break;
+        default:
+            System.out.println("Input error");
+        }
+    }
+
+    // Handle single command
+    public static void handleSingleCommand(String curCommand) {
+        switch (curCommand) {
+        case "list":
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i < tasks.size(); i++) {
+                System.out.println((i + 1) + "." + MessageManager.taskToString(tasks.get(i)));
+            }
+            break;
+        case "bye":
+            System.out.println("Bye. Hope to see you again soon!");
+            break;
+        case "help":
+            MessageManager.printHelp("help");
+            break;
+        default:
+            System.out.println("Input error");
+        }
+    }
+
+    // Checks if a string is a positive numeric value
+    public static int getPositiveNumeric(String str) {
+        try {
+            return Integer.parseInt(str);
+        } catch(NumberFormatException e){
+            return -1;
+        }
+    }
+
+    // Write into file
+    public static void writeIntoFile() {
+        try {
+            String dir = System.getProperty("user.dir") ;
+            File newDir = new File(dir + "\\data");
+            newDir.mkdir();
+
+            File newFile = new File(dir + "\\data\\duke.txt");
+            newFile.createNewFile();
+
+            FileWriter myWriter = new FileWriter(dir + "\\data\\duke.txt");
+
+            for (Task t : tasks) {
+                myWriter.write(t.getTask()
+                        + " / " + t.getTrail()
+                        + " / " + t.getType()
+                        + " / " + (t.isDone() ? 1 : 0)
+                        + "\n"
+                );
+            }
+
+            myWriter.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    // Read from file
+    public static void readFromFile() {
+        try {
+            String dir = System.getProperty("user.dir");
+            File file = new File(dir + "\\data\\duke.txt");
+            file.createNewFile();
+
+            FileReader fileReader = null;
+            fileReader = new FileReader(file);
+
+            BufferedReader bufferedreader = new BufferedReader(fileReader);
+            String strLine;
+
+            while ((strLine = bufferedreader.readLine()) != null) {
+                String[] saved = strLine.split(" / ", 4);
```

Comment:
Avoid using magic numbers like 4 - extract them out to constants and name them appropriately using proper naming conventions